### PR TITLE
Enabled to auto fill requester and processed_by attributes

### PR DESCRIPTION
**Pull request by hsong-rh**
Auto filled attributes of requester and processed_by if they are not set.

### CODE REVIEW

File: app/services/request_create_service.rb [hunk 1]
```diff
@@ -24,6 +24,7 @@ def create(options)
       :stages   => stages
     )
     Request.create!(create_options).tap do |request|
+      request.update_attributes!(:requester => ContextService.new(request.context).identity_fullname) unless request.requester
```
Comment (bzwei) [inline app/services/request_create_service.rb]: remove it. insert before line 26
```
unless options[:requester]
  requester = ManageIQ::API::Common::Request.current.user
  create_options[:requester] = "#{requester.first_name} #{requester.last_name}"
end
```

File: app/services/context_service.rb [hunk 2]
```diff
@@ -13,6 +13,16 @@ def as_org_admin(&block)
     switch_context(org_admin_context, &block)
   end
 
+  def identity_username
```
Comment (bzwei) [inline app/services/context_service.rb]: do not need the two new methods

File: app/services/action_create_service.rb [hunk 3]
```diff
@@ -7,7 +7,8 @@ def initialize(stage_id)
 
   def create(options)
     stage_options = validate_operation(options)
-    Action.create!(options.merge(:stage => stage)).tap do
+    Action.create!(options.merge(:stage => stage)).tap do |action|
```
Comment (bzwei) [inline app/services/action_create_service.rb]: apply the same comment for `request_create_service`

File: app/models/action.rb [hunk 4]
```diff
@@ -8,7 +8,6 @@ class Action < ApplicationRecord
   DENY_OPERATION    = 'deny'.freeze
   OPERATIONS = [NOTIFY_OPERATION, SKIP_OPERATION, MEMO_OPERATION, APPROVE_OPERATION, DENY_OPERATION].freeze
 
-  validates :operation,    :inclusion => { :in => OPERATIONS }
-  validates :processed_by, :presence  => true
+  validates :operation, :inclusion => { :in => OPERATIONS }
```
Comment (bzwei) [inline app/models/action.rb]: no need to remove validates for processed_by

File: spec/requests/v1.0/workflows_spec.rb [hunk 5]
```diff
@@ -8,7 +8,7 @@
 
   let!(:template) { create(:template) }
   let(:template_id) { template.id }
-  let!(:workflows) { create_list(:workflow, 20, :template_id => template.id) }
+  let!(:workflows) { create_list(:workflow, 16, :template_id => template.id) }
```
Comment (bzwei) [inline spec/requests/v1.0/workflows_spec.rb]: why 16 or 20?

Comment (hsong-rh) [reply spec/requests/v1.0/workflows_spec.rb]: 20 exceeds the test boundary due to the extra seeding one

File: spec/services/action_create_service_spec.rb [hunk 6]
```diff
@@ -71,6 +77,13 @@
     end
   end
 
+  context 'auto set processed_by if nil' do
+    it 'create a new action with nil processed_by' do
+      action = svc1.create('operation' => Action::MEMO_OPERATION, 'comments' => 'later')
+      expect(action.processed_by).to eq("jdoe")
```
Comment (bzwei) [inline spec/services/action_create_service_spec.rb]: should not hardcode `jdoe`, should be `ManageIQ::API::Common::Request.current.user.username`

File: spec/services/request_create_service_spec.rb [hunk 7]
```diff
@@ -12,6 +12,20 @@
     end
   end
 
+  context 'with auto fill requester' do
+    it 'auto fill requester if it is nil' do
+      request = subject.create(:name => 'req1', :content => 'test me')
+      request.reload
+      expect(request.requester).to eq("John Doe")
```
Comment (bzwei) [inline spec/services/request_create_service_spec.rb]: same, get the full name from current header

File: spec/services/request_create_service_spec.rb [hunk 8]
```diff
@@ -12,6 +12,20 @@
...
+    end
+
+    it 'skip to auto fill if it is set' do
```
Comment (bzwei) [inline spec/services/request_create_service_spec.rb]: `skips auto filling if requester is set`

File: spec/services/request_create_service_spec.rb (hunk 7 above)
Comment (hsong-rh) [reply spec/services/request_create_service_spec.rb]: There is no `fullname` method in in either `header` or `user`

**Pull request merged**